pkg/controller/wildflyserver/wildflyserver_controller.go

```diff
@@ -539,6 +542,7 @@ func LabelsForWildFly(w *wildflyv1alpha1.WildFlyServer) map[string]string {
 	labels["app.kubernetes.io/name"] = w.Name
 	labels["app.kubernetes.io/managed-by"] = os.Getenv("LABEL_APP_MANAGED_BY")
 	labels["app.openshift.io/runtime"] = os.Getenv("LABEL_APP_RUNTIME")
+	labels["app.kubernetes.io/instance"] = fmt.Sprintf("wildfly-%s", w.GetName())
```

Comment:
I'm a bit confusing with the recommendations for these labels (looking at https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/).

* app.kubernetes.io/name -> The name of the application (eg `mysql`)
* app.kubernetes.io/instance -> A unique name identifying the instance of an application (e.g. `mysql-abcxzy`)

The key difference being that an application `name` can be installed many times in the same cluster (or namespace) while an `instance` is unique.

The existing `app.kubernetes.io/name` matches the CRD name (`w.Name` which is the same that `w.GetName()` and is thus already unique in the namespace (which is a stronger semantic that the recommendations from kubernetes.io). We could use the same value for the  `app.kubernetes.io/instance` without losing any information.

Reply:
Hmm... yes that is confusing, I haven't noticed that.

This explains that an application can have multiple instances on the same namespace:
https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/#applications-and-instances-of-applications

That got me thinking,
maybe we shouldn't use the `app.kubernetes.io/instance` label,
or any other recommended label for that manner, as the selector label.
(maybe we should use the `instance` label regardless of the selector label, but that's for another discussion). 

To make stuff more robust, maybe we should use our own custom label,
something in the sorts of:

```go
labels[fmt.Sprintf("%s/%s", os.Getenv("LABEL_APP_RUNTIME"), "selector")] = fmt.Sprintf("%s-%s", os.Getenv("LABEL_APP_RUNTIME"), w.Name)
// or simply
labels[fmt.Sprintf("%s/%s", os.Getenv("LABEL_APP_RUNTIME"), "selector")] = w.Name
```

Either that, or we can just use the existing `app.kubernetes.io/name` as you suggested.
What do you think?

Reply:
Implemented the changes request here: https://github.com/wildfly/wildfly-operator/pull/210/commits/94690ba53d1a12af15706f152a3bdf57de3a0fc9

The existing `app.kubernetes.io/name` label, valued with the unique CR name,
is now also being used for the HPA as the selector label.